cluster-provision/images/vm-image-builder/README.md

```diff
@@ -0,0 +1,162 @@
+# Customize ephemeral container-disk images for KubeVirt VM's
+
+This tool is targeted for KubeVirt developers, users or anyone who
+would like to create KubeVirt VM's with customized container-disk,
+without the hassle of installing and customizing the OS manually.
+
+Use cases:
+- VM's for Demos
+- Pre-configured Kubevirt VM image for tests, for example:
+    - Image with qemu-guest-agent installed and configured.
+    - Image with sriov drivers for sriov-lane images.
+    - Image with DPDK package for testing DPDK applications on KubeVirt.
+
+## Create customized container-disk
+
+In order to create customized  container-disk for KubeVirt VM,
+use `create-containderdisk.sh` script.
+
+This script automates the process of customizing VM image and build Kubevirt VM container-disk-image.
+Soon there will be used for creating and image automatically using CI.
+
+What this script does:
+- Download VM image from given URL
+
+- Customize VM image using `customize-image.sh` script, according to given cloud-config file.
+  Keeping customizing image method loosely-coupled, so it will be easy to maintain.
+  You can use any image customizing script by exporting `CUSTOMIZE_IMAGE_SCRIPT=<script path>`.
+
+- Building container-disk image using `build-containerdisk.sh`
+  according to the doc:
+  https://github.com/kubevirt/kubevirt/blob/master/docs/container-register-disks.md
+
+- Store the image in local registry and exports the container image to .tar file, so it will be easier to store or send.
+
+This script expected variables:  
+- `VM_IMAGE_URL` - URL to download the image
+- `CLOUD_CONFIG_PATH` - cloud-conifg file path used  for customizing the image with cloud-init
+- `IMAGE_NAME`, `TAG` - container-disk image tag that will be created tag details `IMAGE_NAME:TAG`
+- `OS_VARIANT`- required by customize-image.sh
+
+Notes:
+- If no variables passed, the script will create container-disk image according to the files at `example/`.
+- Currently, image customization process done by spinning up live VM with cloud-init disk to provision the image with the changes described by the cloud-config file.
+- It is necessary to add `sudo shutdown` at the end of the cloud-config`runcmd` block, so the script will not wait for user input.
+- You can use your own image customizing script by exporting the path to `CUSTOMIZE_IMAGE_SCRIPT`.
+
+This script requires the following packages:
+- cloud-utils
+- docker-ce
```

Comment:
You should consider changing to podman in a following change.

Reply:
podman can be used as well I will add it to README

Reply:
I will test it and consider a following PR